### ISSUE TITLE
GPUImageMovieWriter cannot save audio/sound

### ISSUE THREAD

**Issue by tryagaino** [BradLarson/GPUImage#2417]
A normal video taken from iPhone, after applied the filter, saved with GPUImageMovieWriter, but the result video has no audio/sound, even used the shouldPassthroughAudio and audioEncodingTargert as example did. 

`self.gpuMovie = [[GPUImageMovie alloc] initWithURL:self.videoURL];
   
    self.gpuFilter = [[GPUImageGrayscaleFilter alloc] init];
    NSString *finalMoviePath = [NSTemporaryDirectory() stringByAppendingPathComponent:@"filter.mov"];
    unlink([finalMoviePath UTF8String]);
    NSURL *finalMovieURL = [NSURL fileURLWithPath:finalMoviePath];
    self.gpuWriter = [[GPUImageMovieWriter alloc] initWithMovieURL:finalMovieURL size:CGSizeMake(224, 128)];  
    [self.gpuMovie addTarget:self.gpuFilter];
    [self.gpuFilter addTarget:self.gpuWriter];
    
    self.gpuWriter.shouldPassthroughAudio = YES;
    self.gpuMovie.audioEncodingTarget = self.gpuWriter;

    [self.gpuMovie enableSynchronizedEncodingUsingMovieWriter:self.gpuWriter];
    [self.gpuWriter startRecording];
    [self.gpuMovie startProcessing];

    __weak typeof(self) wself = self;
    [self.gpuWriter setCompletionBlock:^{
        NSLog(@"Finally, Done");
        [wself.gpuFilter removeAllTargets];
        [wself.gpuWriter finishRecording];
        `

I am using iOS 10. Any help will be appreciate.

**Comment by tryagaino** [comment]
I am so stupid. It seems another typical issue when you try to run the code inside iPhone simulator. Although I don't know why iPhone simulator will cause this problem, but when I finally decided to try on a real iPhone, it worked.
I cannot believe that I wasted whole day for this issue, I guess I totally forgot it because the video part (and other AVFoundation code I used) all worked smoothly even in simulator.
Anyway, hope other people never made such stupid mistake like I did.
